### PR TITLE
CI with GitHub actions #145

### LINKED ISSUE #145: Move CI to GitHub Actions
It would be cool if we could also build maven releases. 🤔

### PR DESCRIPTION
**Pull request by subzero10**
STATUS
**READY**

Fixes #145.
What the title says :)

TODO:

- [x] Create workflow
- [x] Uncomment enforcer plugin
- [x] Pass CI
- [x] Test with failing test: [workflow run](https://github.com/honeybadger-io/honeybadger-java/actions/runs/2806411338)

### CODE REVIEW

File: pom.xml
Comment (subzero10) [inline pom.xml:130]: Had to downgrade this one.

**Review by joshuap — APPROVED** [review]
I mean, hardly any diff... this looks great to me. :grin: :100:

I am going to ask @JasonTrue (who has maintained this package in the past) to also review before merging this;  he may have some additional followup changes to make. @JasonTrue feel free to merge this if you think it looks good.

**Comment by subzero10** [comment]
Please note that I did not include [this block](https://github.com/honeybadger-io/honeybadger-java/blob/main/.travis.yml#L25) in the github workflow. I planned to do it, but the workflow works without it so I decided not to add it (plus I don't really know what most of that does).

**Pull request merged**
